test/unit/services/test_manufacturer.py

```diff
@@ -2,237 +2,382 @@
+
+        # Ensure updated with expected data
+        self.mock_manufacturer_repository.update.assert_called_once_with(
+            self._updated_manufacturer_id, self._expected_manufacturer_in
+        )
+
+        assert self._updated_manufacturer == self._expected_manufacturer_out
+
+    def check_update_failed_with_exception(self, message: str):
+        """
+        Checks that a prior call to `call_update_expecting_error` worked as expected, raising an exception with the
+        correct message.
+
+        :param message: Message of the raised exception.
+        """
+        self.mock_manufacturer_repository.update.assert_not_called()
+        assert str(self._update_exception.value) == message
+
+
+class TestUpdate(UpdateDSL):
+    """
+    Tests for updating a manufacturer.
+    """
+
+    def test_update(self):
+        """
+        Test updating all fields of a manufacturer.
+        """
+        manufacturer_id = str(ObjectId())
+
+        self.mock_update(
+            manufacturer_id,
+            manufacturer_patch_data=MANUFACTURER_POST_DATA_B,
+            stored_manufacturer_post_data=MANUFACTURER_POST_DATA_A,
+        )
+        self.call_update(manufacturer_id)
+        self.check_update_success()
+
+    def test_update_address_only(self) -> None:
+        """
+        Test updating manufacturer's address only (code should not need regenerating as name doesn't change).
+        """
+        manufacturer_id = str(ObjectId())
+
+        self.mock_update(
+            manufacturer_id,
+            manufacturer_patch_data=MANUFACTURER_POST_DATA_B["address"],
+            stored_manufacturer_post_data=MANUFACTURER_POST_DATA_A,
+        )
+        self.call_update(manufacturer_id)
+        self.check_update_success()
+
+    def test_update_with_non_existent_id(self):
+        """
+        Test updating a manufacturer with a non-existent ID.
+        """
+        manufacturer_id = str(ObjectId())
+
+        self.mock_update(
+            manufacturer_id, manufacturer_patch_data=MANUFACTURER_POST_DATA_A, stored_manufacturer_post_data=None
+        )
+        self.call_update_expecting_error(manufacturer_id, MissingRecordError)
+        self.check_update_failed_with_exception(f"No manufacturer found with ID: {manufacturer_id}")
+
+
+class DeleteDSL(ManufacturerServiceDSL):
+    """
+    Base class for delete tests.
+    """
+
+    _delete_manufacturer_id: str
+
+    def call_delete(self, manufacturer_id: str) -> None:
+        """
+        Calls the `ManufacturerService` `delete` method.
+        :param manufacturer_id: ID of the manufacturer to be deleted.
+        """
+        self._delete_manufacturer_id = manufacturer_id
+        self.manufacturer_service.delete(manufacturer_id)
+
+    def check_delete_success(self) -> None:
+        """
+        Checks that a prior call to `call_delete` worked as expected.
+        """
+        self.mock_manufacturer_repository.delete.assert_called_once_with(self._delete_manufacturer_id)
```

Comment:
```suggestion
        """Checks that a prior call to `call_delete` worked as expected."""
        
        self.mock_manufacturer_repository.delete.assert_called_once_with(self._delete_manufacturer_id)
```
There are some more of these.